messenger/message-recorder.rst

```diff
@@ -0,0 +1,74 @@
+.. index::
+    single: Messenger; Record messages
+
+Record Events Produced by a Handler
```

Comment:
Might be better to find a name that describes the problem than the technical solution. Something like "Event recorder: Tolerating failures while in Doctrine transactions"?